nixos/lib/make-zfs-image.nix

```diff
@@ -219,23 +219,12 @@ let
       nixpkgs-fmt $out
     '';
 
-  mergedConfig =
-    if configFile == null
-    then fileSystemsCfgFile
-    else
-      pkgs.runCommand "configuration.nix" {
-        buildInputs = with pkgs; [ nixpkgs-fmt ];
-      }
-        ''
-          (
-            echo '{ imports = ['
-            printf "(%s)\n" "$(cat ${fileSystemsCfgFile})";
-            printf "(%s)\n" "$(cat ${configFile})";
-            echo ']; }'
-          ) > $out
-
-          nixpkgs-fmt $out
-        '';
+  mergedConfig = pkgs.mergeNixOSModules {
+      name = "configuration.nix";
+      moduleFiles = [
+      fileSystemsCfgFile
+    ] ++ lib.optional (configFile != null) configFile;
```

Comment:
(Also might wanna fixup the formatting)

Reply:
Weird, this is exactly as nixpkgs-fmt formatted it.

Reply:
Just kidding, vscode was hiding part of the whitespace diff.